pyccel/parser/semantic.py

```diff
@@ -1210,7 +1210,7 @@ def _ensure_infered_type_matches_existing(self, dtype, d_var, var, is_augassign,
                     self._current_fst_node.col_offset),
                         severity='error', symbol=var.name)
 
-        elif var.is_ndarray and var.is_alias:
+        elif var.is_ndarray and var.is_alias and not is_augassign:
             # we allow pointers to be reassigned multiple times
             # pointers reassigning need to call free_pointer func
             # to remove memory leaks
```

Comment:
Maybe i'm missing something but why do we need to free aliases ?

Reply:
Aliases contain shape and strides arrays which are independent of the original array. This allows you to do:
```
a = np.ones(4,6)
b = a[:2,:3]
a.shape # 4,6
b.shape # 2,2
```
These arrays need freeing when the alias goes out of scope

Reply:
Oh i see so we free the shape and strides in that case makes sense.

Reply:
Well it's seems fine to me then.

Reply:
In that case, you should `approve` the PR